keras/preprocessing/text_dataset.py

```diff
@@ -153,30 +153,66 @@ def text_dataset_from_directory(directory,
         f'When passing `label_mode="binary"`, there must be exactly 2 '
         f'class_names. Received: class_names={class_names}')
 
-  file_paths, labels = dataset_utils.get_training_or_validation_split(
-      file_paths, labels, validation_split, subset)
-  if not file_paths:
-    raise ValueError(f'No text files found in directory {directory}. '
-                     f'Allowed format: .txt')
-
-  dataset = paths_and_labels_to_dataset(
-      file_paths=file_paths,
-      labels=labels,
-      label_mode=label_mode,
-      num_classes=len(class_names),
-      max_length=max_length)
-  dataset = dataset.prefetch(tf.data.AUTOTUNE)
-  if batch_size is not None:
-    if shuffle:
-      # Shuffle locally at each iteration
-      dataset = dataset.shuffle(buffer_size=batch_size * 8, seed=seed)
-    dataset = dataset.batch(batch_size)
+  if subset == "both":
+    file_paths_train, labels_train = dataset_utils.get_training_or_validation_split(
+        file_paths, labels, validation_split, subset)
+    file_paths_val, labels_val = dataset_utils.get_training_or_validation_split(
+        file_paths, labels, validation_split, subset)
+    if not file_paths_train or file_paths_val:
+      raise ValueError(f'No text files found in directory {directory}. '
```

Comment:
What if only one of the two datasets is empty? In what situation could that occur? Could we provide a better message in that case?

Reply:
I believe it would only occur when validation_split is zero or one. Will fix it